embeddings/pipeline/flair_hps_pipeline.py

```diff
@@ -42,75 +46,73 @@
 ):
     input_column_name: str
     target_column_name: str
 
 
 @dataclass
 class _OptimizedFlairPairClassificationPipelineBase(
-    _HuggingFaceOptimizedPipelineBase[ConfigSpace],
-    ABC,
-    Generic[ConfigSpace],
+    _HuggingFaceOptimizedPipelineBase[ConfigSpace], ABC, Generic[ConfigSpace]
 ):
     input_columns_names_pair: Tuple[str, str]
     target_column_name: str
 
 
 @dataclass
-class _OptimizedFlairClassificationPipelineDefaultsBase(
-    _HuggingFaceOptimizedPipelineDefaultsBase,
-    ABC,
-):
+class _OptimizedFlairPipelineDefaultsBase(_HuggingFaceOptimizedPipelineDefaultsBase, ABC):
     tmp_dataset_dir: TemporaryDirectory[str] = field(init=False, default_factory=TemporaryDirectory)
     tmp_model_output_dir: TemporaryDirectory[str] = field(
         init=False, default_factory=TemporaryDirectory
     )
 
-    @staticmethod
-    def _pop_sampled_parameters(
-        parameters: SampledParameters,
-    ) -> Tuple[str, str, Dict[str, ParameterValues], Dict[str, ParameterValues]]:
-        embedding_name = parameters["embedding_name"]
-        assert isinstance(embedding_name, str)
-        document_embedding = parameters["document_embedding"]
-        assert isinstance(document_embedding, str)
-        task_train_kwargs = parameters["task_train_kwargs"]
-        assert isinstance(task_train_kwargs, dict)
-        load_model_kwargs = parameters["load_model_kwargs"]
-        assert isinstance(load_model_kwargs, dict)
-        return embedding_name, document_embedding, task_train_kwargs, load_model_kwargs
-
 
 @dataclass
 class OptimizedFlairClassificationPipeline(
     OptunaPipeline[
         TextClassificationConfigSpace,
         FlairClassificationPipelineMetadata,
         FlairEvaluationPipelineMetadata,
+        str,
+        datasets.DatasetDict,
+        Corpus,
     ],
     AbstractHuggingFaceOptimizedPipeline[TextClassificationConfigSpace],
-    _OptimizedFlairClassificationPipelineDefaultsBase,
-    _OptimizedFlairClassificationPipelineBase[TextClassificationConfigSpace],
+    _OptimizedFlairPipelineDefaultsBase,
+    _OptimizedFlairPipelineBase[TextClassificationConfigSpace],
 ):
-    def __post_init__(self) -> None:
-        self.dataset_path = Path(self.tmp_dataset_dir.name).joinpath("ds.pkl")
-        # Type: ignore is temporal solution due to issue #152 https://github.com/CLARIN-PL/embeddings/issues/152
-        super().__init__(
-            preprocessing_pipeline=FlairTextClassificationPreprocessingPipeline(
-                dataset_name=self.dataset_name,
+    def _init_dataset_path(self) -> None:
+        self.dataset_path: T_path
+        if self.ignore_preprocessing_pipeline:
+            self.dataset_path = Path(self.dataset_name_or_path)
+            if not self.dataset_path.exists():
+                raise FileNotFoundError("Dataset path not found")
+        else:
+            self.dataset_path = Path(self.tmp_dataset_dir.name).joinpath("ds.pkl")
```

Comment:
This method is repeated in multiple places. Maybe it should be incorporated in `_OptimizedFlairPipelineBase` ?

Reply:
tmp_dataset_dir is defined in _OptimizedFlairPipelineDefaultsBase class while abstract methods are defined all in AbstractHuggingFaceOptimizedPipeline class; it is true that this method is repeated twice in this file but this method is also different for Classification/PairClassification and SequenceLabeling. Currently, I somehow tried to keep the dataclass attributes and abstractmethods separately which IMHO improves the readability of the code at the expense of code duplication. Although you are right that this code is repeated; I introduced a new abstract class to remove 2 repeated methods in Classification/PairClassification pipelines: _init_dataset_path and pop_sampled_parameters.
Indeed, the HPS pipelines need thorough refactor which will be addressed in further issues/PRs.